### PR TITLE
Fix Gradle options for local/non-CI builds

### PR DESCRIPTION
**Pull request by PawelLipski**
As suggested by @mkondratek on #275

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -79,6 +81,9 @@ subprojects {
     tasks.withType(JavaCompile) {
       // Treat each compiler warning (esp. the ones coming from Checker Framework) as an error.
       options.compilerArgs << '-Werror'
+      if (System.getenv('CI') != 'true') {
+        options.forkOptions.jvmArgs += ['-Xmx12g', '-XX:+HeapDumpOnOutOfMemoryError']
```
Comment (PawelLipski) [inline build.gradle:85]: Skipping `-XX:MaxPermSize` - deprecated since Java 8

Comment (mkondratek) [reply build.gradle:85]: IMHO this (local options) should be mentioned in the contribution guidelines anyway

Comment (PawelLipski) [reply build.gradle]: @mkondratek I reworked that slightly (also decreased the default limit, 12GB was kinda a looot), pls see the updated version

Comment (mkondratek) [reply build.gradle]: this is fine :fire:

**Pull request merged**